From fb841ede3d3752174fe07fd1130f672f1762989e Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Wed, 14 Sep 2016 15:09:34 +0200
Subject: [PATCH] libfc: Fix PRLI response handling yet again
Patch-Mainline: submitted linux-scsi 2016/09/12
References: bsc#998385

Even with the latest fixes we still don't handle everything
correct; the service params are for the remote port, so we
don't actually care if the ->prli callout works or not.

Reported-by: Martin Wilck <martin.wilck@suse.com>
Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/libfc/fc_rport.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/drivers/scsi/libfc/fc_rport.c b/drivers/scsi/libfc/fc_rport.c
index e94a56e..d1e95ef 100644
--- a/drivers/scsi/libfc/fc_rport.c
+++ b/drivers/scsi/libfc/fc_rport.c
@@ -1181,17 +1181,15 @@ static void fc_rport_prli_resp(struct fc_seq *sp, struct fc_frame *fp,
 		prov = fc_passive_prov[rdata->spp_type];
 		if (prov) {
 			memset(&temp_spp, 0, sizeof(temp_spp));
-			resp_code = prov->prli(rdata, pp->prli.prli_spp_len,
-					       &pp->spp, &temp_spp);
-		} else {
-			memcpy(&temp_spp, &pp->spp, sizeof(temp_spp));
+			prov->prli(rdata, pp->prli.prli_spp_len,
+				   &pp->spp, &temp_spp);
 		}
 		/*
 		 * Check if the image pair could be established
 		 */
 		if (rdata->spp_type != FC_TYPE_FCP ||
 		    resp_code != FC_SPP_RESP_ACK ||
-		    !(temp_spp.spp_flags & FC_SPP_EST_IMG_PAIR)) {
+		    !(pp->spp.spp_flags & FC_SPP_EST_IMG_PAIR)) {
 			/*
 			 * Nope; we can't use this port as a target.
 			 */
